cmd/kube-apiserver/app/options/options.go

```diff
@@ -109,6 +110,65 @@ func NewServerRunOptions() *ServerRunOptions {
 
+// ComponentConfigz produces a componentconfig object from server options.
+func (s *ServerRunOptions) ComponentConfigz() componentconfig.APIServerConfiguration {
+	serviceClusterIPRange := s.ServiceClusterIPRange.String()
+	serviceNodePortRange := s.ServiceNodePortRange.String()
+
+	componentconfig := componentconfig.APIServerConfiguration{
+		AdmissionControl:            s.GenericServerRunOptions.AdmissionControl,
+		AdmissionControlConfigFile:  s.GenericServerRunOptions.AdmissionControlConfigFile,
+		AdvertiseAddress:            s.GenericServerRunOptions.AdvertiseAddress.String(),
+		CorsAllowedOriginList:       s.GenericServerRunOptions.CorsAllowedOriginList,
+		ExternalHost:                s.GenericServerRunOptions.ExternalHost,
+		MaxRequestsInFlight:         s.GenericServerRunOptions.MaxRequestsInFlight,
+		MaxMutatingRequestsInFlight: s.GenericServerRunOptions.MaxMutatingRequestsInFlight,
+		MinRequestTimeout:           s.GenericServerRunOptions.MinRequestTimeout,
+		TargetRAMMB:                 s.GenericServerRunOptions.TargetRAMMB,
+		WatchCacheSizes:             s.GenericServerRunOptions.WatchCacheSizes,
+		AuditLogOptions: componentconfig.APIServerAuditLogOptions{
+			Path:       s.Audit.Path,
+			MaxAge:     s.Audit.MaxAge,
+			MaxBackups: s.Audit.MaxBackups,
+			MaxSize:    s.Audit.MaxSize,
+		},
+		EnableProfiling:           s.Features.EnableProfiling,
+		EnableContentionProfiling: s.Features.EnableContentionProfiling,
+		EnableSwaggerUI:           s.Features.EnableSwaggerUI,
+		StorageConfig: componentconfig.APIServerEtcdConfiguration{
+			Type:       s.Etcd.StorageConfig.Type,
+			Prefix:     s.Etcd.StorageConfig.Prefix,
+			ServerList: s.Etcd.StorageConfig.ServerList,
+			KeyFile:    s.Etcd.StorageConfig.KeyFile,
+			CertFile:   s.Etcd.StorageConfig.CertFile,
+			CAFile:     s.Etcd.StorageConfig.CAFile,
+			Quorum:     s.Etcd.StorageConfig.Quorum,
+			DeserializationCacheSize: s.Etcd.StorageConfig.DeserializationCacheSize,
+		},
+		EtcdServersOverrides:    s.Etcd.EtcdServersOverrides,
+		DefaultStorageMediaType: s.Etcd.DefaultStorageMediaType,
+		DeleteCollectionWorkers: s.Etcd.DeleteCollectionWorkers,
+		EnableGarbageCollection: s.Etcd.EnableGarbageCollection,
+		EnableWatchCache:        s.Etcd.EnableWatchCache,
+		CloudConfigFile:         s.CloudProvider.CloudConfigFile,
+		CloudProvider:           s.CloudProvider.CloudProvider,
+		StorageSerialization: componentconfig.APIServerStorageSerializationOptions{
+			StorageVersions:        s.StorageSerialization.StorageVersions,
+			DefaultStorageVersions: s.StorageSerialization.DefaultStorageVersions,
+		},
+		AllowPrivileged:           &s.AllowPrivileged,
+		KubernetesServiceNodePort: &s.KubernetesServiceNodePort,
+		MasterCount:               &s.MasterCount,
+		MaxConnectionBytesPerSec:  &s.MaxConnectionBytesPerSec,
+		ServiceClusterIPRange:     &serviceClusterIPRange,
+		ServiceNodePortRange:      &serviceNodePortRange,
+	}
+
+	componentconfig.EventTTL.Duration = s.EventTTL
```

Comment:
So there are 120 - apiserver args that I think need to be audited on what can be exposed.  Ideally I would like more etcd data exposed but possibly only via a service network.  At some point operators will want to have a service network that can be accessed and doesn't lie on the same plane as everything else, which should help to alleviate some of the security concerns.